plugins/chromatic_correction.py

```diff
@@ -0,0 +1,252 @@
+            # Extract the matching points
+            points1 = centers_ref[corresponding_pairs[:, 0]]
+            points2 = centers_new[corresponding_pairs[:, 1]]
+
+            if len(corresponding_pairs) < 4:
+                txt = ('The chromatic correction did not happen for channel with %s. Skipping this channel.\n\n'
+                       'Alternatively, navigate to location with more beads or change the bead diameter in the script'
+                       % im.metadata[model.MD_DESCRIPTION])
+                box = wx.MessageDialog(self.main_app.main_frame,
+                                       txt, "Partial failure of chromatic correction", wx.OK)
+                box.ShowModal()
+                box.Destroy()
+
+            else:
+                # Estimate the transformation matrix
+                affine = AffineTransform.from_pointset(points1, points2)
+                mat = numpy.eye(3)
+                mat[:2, :2] = affine.matrix
+                mat[:2, 2] = affine.translation
+                ref_mat = numpy.eye(3)
+                mat = ref_mat @ mat
+
+                # Scale is the norm of the first two columns
+                sx = numpy.linalg.norm(mat[:2, 0])
+                sy = numpy.linalg.norm(mat[:2, 1])
+                scale2 = numpy.array([sx, sy])
+
+                # Compute rotation
+                rotation = numpy.arctan2(mat[1, 0] / sx, mat[0, 0] / sx)
+
+                # compute scale
+                scale = (1 / scale2[0], 1 / scale2[1])
+
+                # compute shear
+                a, b = mat[0, 0], mat[0, 1]
+                c, d = mat[1, 0], mat[1, 1]
+                shear = numpy.arctan2(a * c + b * d, scale[0] * scale[1])
+
+                # compute translation
+                tx = mat[0, 2]
+                ty = mat[1, 2]
+                pixel_size = im.metadata[model.MD_PIXEL_SIZE]
+                translation = (tx * pixel_size[0], -ty * pixel_size[1])
+
+                # stream settings
+                input_wl = im.metadata[model.MD_IN_WL]
+                output_wl = im.metadata[model.MD_OUT_WL]
+
+                try:
+                    fl_band = im.metadata[model.MD_EXTRA_SETTINGS]['Filter Wheel']['position'][0]['band']
+                    # The below metadata dictionary will be added to the microscope file using input command
+                    # under the filter wheel component, only if the metadata has Pixel size cor, Centre position cor,
+                    # Rotation cor, and Shear cor for a minimum one filter wheel band position
+                    trans_per_channel[model.MD_CHROMATIC_COR][fl_band] = {
+                        "Pixel size cor": [float(scale[0]), float(scale[1])],
+                        "Centre position cor": [float(translation[0]), float(translation[1])],
+                        "Rotation cor": float(rotation),
+                        "Shear cor": float(shear),
+                        # Extra information saved for tracking/debugging purpose
```

Comment:
The drawback with such "debugging" metadata is that it'll be on the microscope file, so the MD updater would read it, and copy it onto the other components (assuming you keep the code generic/simple). I think it'd be best if you could put them as comment, or at least as special metadata which will not be confused.